views/js/controller/runner/runner.js

```diff
@@ -44,7 +46,8 @@ define([
      */
 
     runner.registerProvider('qti', qtiProvider);
-    proxy.registerProxy('qtiServiceProxy', qtiServiceProxy);
+    proxy.registerProvider('qtiServiceProxy', qtiServiceProxy);
+    communicator.registerProvider('poll', pollProvider);
```

Comment:
Shouldn't you register the provider in the proxy itself ? This is the responsibility of the proxy to select the way to communicate.